# Health Insurance Predictions

## Business Question
"How do personal and lifestyle factors such as age, BMI, and smoking habits impact medical insurance charges, and how can this information be used to develop risk-based pricing strategies?"

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import OneHotEncoder, StandardScaler

In [13]:
df = pd.read_csv('insurance.csv')
df

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [15]:
df.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

In [16]:
df.shape

(1338, 7)

The dataset contains:
- 1338 rows and 7 columns

## Data Cleaning

In [17]:
df.isna().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

In [18]:
# checking duplicates
df.duplicated().sum()

1

In [19]:
# dropping duplicates
df = df.drop_duplicates()

In [21]:
# encoding the categorical features
categorical = ['sex', 'smoker', 'region']

encoder = OneHotEncoder(drop='first', sparse=False)

# fit andd transform the categorical columns
encoded_array = encoder.fit_transform(df[categorical])

# encoded feature names
encoded_cols = encoder.get_feature_names(categorical)

In [22]:
# convert to dataframe
encoded_df = pd.DataFrame(encoded_array, columns=encoded_cols, index=df.index)

# Drop original categorical columns and concatenate the encoded DataFrame
df_encoded = pd.concat([df.drop(columns=categorical), encoded_df], axis=1)

In [23]:
df_encoded

age     bmi  children      charges  sex_1  smoker_1  region_1  region_2  \
0      19  27.900         0  16884.92400    0.0       1.0       0.0       0.0   
1      18  33.770         1   1725.55230    1.0       0.0       0.0       1.0   
2      28  33.000         3   4449.46200    1.0       0.0       0.0       1.0   
3      33  22.705         0  21984.47061    1.0       0.0       1.0       0.0   
4      32  28.880         0   3866.85520    1.0       0.0       1.0       0.0   
...   ...     ...       ...          ...    ...       ...       ...       ...   
1333   50  30.970         3  10600.54830    1.0       0.0       1.0       0.0   
1334   18  31.920         0   2205.98080    0.0       0.0       0.0       0.0   
1335   18  36.850         0   1629.83350    0.0       0.0       0.0       1.0   
1336   21  25.800         0   2007.94500    0.0       0.0       0.0       0.0   
1337   61  29.070         0  29141.36030    0.0       1.0       1.0       0.0   

      region_3  
0          1.0  
1          0.0  
2          0.0  
3          0.0  
4          0.0  
...        ...  
1333       0.0  
1334       0.0  
1335       0.0  
1336       1.0  
1337       0.0  

[1337 rows x 9 columns]